### PR TITLE
add a new config for translation, tests, render in log

### PR DESCRIPTION
**Pull request by smitsanghvi21**
- Added `remoteTranslations ` in config under model.
- A test to parse the config.
- Be able to set its value in kibble.json
- Render the value in the log.

### CODE REVIEW

File: kibble/models/config.go [hunk 1]
```diff
@@ -33,6 +33,7 @@ type Config struct {
 	DefaultTimeZone           string                    `json:"defaultTimeZone"`
 	DefaultDateFormat         string                    `json:"defaultDateFormat"`
 	DefaultTimeFormat         string                    `json:"defaultTimeFormat"`
+	RemoteTranslations 		  bool						`json:"remoteTranslations"`
```
Comment (PeculiarGoat) [inline kibble/models/config.go:36]: Too many spaces!  If you get the Go extension for VSCode then it should automatically format these things nicely.

File: kibble/render/render.go [hunk 2]
```diff
@@ -69,7 +69,8 @@ func Watch(sourcePath string, buildPath string, cfg *models.Config, port int32,
 
 // Render - render the files
 func Render(sourcePath string, buildPath string, cfg *models.Config) int {
-
+	
+	fmt.Println("remoteTranslations is: ",cfg.RemoteTranslations)
```
Comment (PeculiarGoat) [inline kibble/render/render.go:73]: I wonder if `log.Infof("remoteTranslations: %t", cfg.RemoteTranslations)` would be better for consistency.

**Review by PeculiarGoat — CHANGES_REQUESTED** [review]
You should also add an `## Unreleased` header at the top of the changelog and an entry for this PR.

**Review by PeculiarGoat — APPROVED** [review]
Nice work!

@sarge are you happy with this?

File: kibble/models/config.go [hunk 3]
```diff
@@ -33,6 +33,7 @@ type Config struct {
 	DefaultTimeZone           string                    `json:"defaultTimeZone"`
 	DefaultDateFormat         string                    `json:"defaultDateFormat"`
 	DefaultTimeFormat         string                    `json:"defaultTimeFormat"`
+	RemoteTranslations 	  bool			    `json:"remoteTranslations"`
```
Comment (stajs) [inline kibble/models/config.go:36]: Not a fan of the wording - can we word as yes/no statement?

```suggestion
	useTranslationsApi 	  bool			    `json:"useTranslationsApi"`
```

File: kibble/config/config_test.go [hunk 4]
```diff
@@ -41,3 +39,9 @@ func TestLoadLanguagesConfigWithObjects(t *testing.T) {
 	assert.Equal(t, cfg.Languages["it"].Code, "it_IT")
 	assert.Equal(t, cfg.Languages["it"].Name, "Italian")
 }
+
+func TestLoadConfig(t *testing.T) {
+	sitePath = "../sample_site/kibble.json"
+	cfg := LoadConfig(false, "xxx", false)
```
Comment (stajs) [inline kibble/config/config_test.go:45]: Just for people wondering what `xxx` is...

```suggestion
	cfg := LoadConfig(false, "api key", false)
```

File: kibble/sample_site/kibble.json [hunk 5]
```diff
@@ -47,5 +47,6 @@
             "pageSize": 0,
             "defaultLanguageOnly": true
           }
-    ]
+    ],
+    "remoteTranslations":true
```
Comment (stajs) [inline kibble/sample_site/kibble.json:51]:
```suggestion
    "remoteTranslations": true
```

File: changelog.md [hunk 6]
```diff
@@ -1,5 +1,8 @@
 # Change log
 
+## Unreleased
+Config for Custom translations- remoteTranslations
```
Comment (stajs) [inline changelog.md:4]:
```suggestion
Config for retrieving translations from a new Languages API - `useTranslationsAPI`. This API is not yet available but adding a feature toggle in advance for development.
```

File: kibble/config/config_test.go [hunk 7]
```diff
@@ -41,3 +39,9 @@ func TestLoadLanguagesConfigWithObjects(t *testing.T) {
...
+	assert.Equal(t, true, cfg.RemoteTranslations)
```
Comment (stajs) [inline kibble/config/config_test.go:46]: Don't forget we need tests to hit the AC's for the negative cases - false and missing. We probably need independent `kibble.json` for each and then you should be able to set the `sitePath` for each test.

**Review by stajs — CHANGES_REQUESTED** [review]

File: changelog.md (hunk 6 above)
Comment (stajs) [reply changelog.md:4]: Don't be afraid to elaborate the changes for external audiences that don't have our knowledge - Kibble is a public repo.

File: changelog.md [hunk 8]
```diff
@@ -1,7 +1,7 @@
 # Change log
 
 ## Unreleased
-Config for Custom translations- remoteTranslations
+Config for retrieving translations from a new Languages API - `useTranslationsAPI`. This API is not yet available but adding a feature toggle in advance for development.
```
Comment (stajs) [inline changelog.md:4]: My bad for typo....

```suggestion
Config for retrieving translations from a new Translations API - `useTranslationsAPI`. This API is not yet available but adding a feature toggle in advance for development.
```

File: kibble/config/config_test.go [hunk 9]
```diff
@@ -41,3 +39,23 @@ func TestLoadLanguagesConfigWithObjects(t *testing.T) {
...
+	cfg := LoadConfig(false, "api-key", false)
+	assert.Equal(t, false, cfg.UseTranslationsApi)
+}
+
+func TestLoadConfigMissing(t *testing.T) {
+	sitePath = "./test/kibble-translations-key-missing.json"
+	cfg := LoadConfig(false, "api-key", false)
+	if(!cfg.UseTranslationsApi){
+		log.Infof("No config")
+	}
```
Comment (stajs) [inline kibble/config/config_test.go:60]: Don't worry about logging, just need the assert.

**Review by stajs — CHANGES_REQUESTED** [review]
Looking good, couple of minor things.

Are you able to pare back the configs to just the `useTranslationsApi` key, or the bare minimum?

**Review by sarge — COMMENTED** [review]
Looks good I agree with @stajs

File: kibble/config/config_test.go [hunk 10]
```diff
@@ -41,3 +39,21 @@ func TestLoadLanguagesConfigWithObjects(t *testing.T) {
 	assert.Equal(t, cfg.Languages["it"].Code, "it_IT")
 	assert.Equal(t, cfg.Languages["it"].Name, "Italian")
 }
+
+func TestLoadConfig(t *testing.T) {
```
Comment (stajs) [inline kibble/config/config_test.go:43]:
```suggestion
func TestUseTranslationsApiTrue(t *testing.T) {
```

File: kibble/config/config_test.go [hunk 11]
```diff
@@ -41,3 +39,21 @@ func TestLoadLanguagesConfigWithObjects(t *testing.T) {
...
+	sitePath = "./test/kibble-translations-key-true.json"
+	cfg := LoadConfig(false, "api-key", false)
+	assert.Equal(t, true, cfg.UseTranslationsApi)
+}
+
+func TestLoadConfigFalse(t *testing.T) {
```
Comment (stajs) [inline kibble/config/config_test.go:49]:
```suggestion
func TestUseTranslationsApiFalse(t *testing.T) {
```

File: kibble/config/config_test.go [hunk 12]
```diff
@@ -41,3 +39,21 @@ func TestLoadLanguagesConfigWithObjects(t *testing.T) {
...
+	sitePath = "./test/kibble-translations-key-false.json"
+	cfg := LoadConfig(false, "api-key", false)
+	assert.Equal(t, false, cfg.UseTranslationsApi)
+}
+
+func TestLoadConfigMissing(t *testing.T) {
```
Comment (stajs) [inline kibble/config/config_test.go:55]:
```suggestion
func TestUseTranslationsApiMissing(t *testing.T) {
```

**Review by stajs — CHANGES_REQUESTED** [review]
Suggestion for better test names

File: kibble/config/config_test.go [hunk 13]
```diff
@@ -41,3 +39,21 @@ func TestLoadLanguagesConfigWithObjects(t *testing.T) {
...
+	sitePath = "./test/kibble-translations-key-missing.json"
+	cfg := LoadConfig(false, "api-key", false)
+	assert.NotContains(t, "useTranslationsApi", cfg.UseTranslationsApi)
```
Comment (PeculiarGoat) [inline kibble/config/config_test.go:58]: Is this test right?  `cfg.UseTranslationsApi` _does_ exist and it should have the boolean zero value (`false`).

Comment (PeculiarGoat) [reply kibble/config/config_test.go:58]: Yeah... https://pkg.go.dev/github.com/stretchr/testify/assert#NotContains is for strings, lists, and maps... not structs!

Comment (PeculiarGoat) [reply kibble/config/config_test.go:58]: Use this instead: https://pkg.go.dev/github.com/stretchr/testify/assert#Zero

Comment (stajs) [reply kibble/config/config_test.go]: Actually, I want this to be `false` (sorry) so that later when we use it in a condition we only have to worry about `true`/`false`. Is that possible?

Perhaps you can [add a default value of `false` before the JSON unmarshal](https://github.com/shift72/kibble/blob/master/kibble/config/config.go#L47-L50)?

File: kibble/config/config_test.go [hunk 14]
```diff
@@ -2,9 +2,8 @@ package config
 
 import (
 	"testing"
-
 	"kibble/models"
-
+	"fmt"
```
Comment (PeculiarGoat) [inline kibble/config/config_test.go:6]: Is this required?

File: kibble/config/config_test.go (hunk 13 above)
Comment (PeculiarGoat) [reply kibble/config/config_test.go]: The boolean zero value is false.

File: kibble/config/config_test.go (hunk 14 above)
Comment (smitsanghvi21) [reply kibble/config/config_test.go:6]: No, had used it for logging the value.

File: kibble/config/config_test.go (hunk 13 above)
Comment (smitsanghvi21) [reply kibble/config/config_test.go]: As Jack mentioned [zero function](https://tour.golang.org/basics/12) takes false by default for boolean by default @stajs

File: kibble/config/config_test.go (hunk 14 above)
Comment (smitsanghvi21) [reply kibble/config/config_test.go:6]: Nope, just had used it for logging.

File: kibble/config/config_test.go (hunk 13 above)
Comment (stajs) [reply kibble/config/config_test.go]: Okay then make it like this:

```
assert.Equal(t, false, cfg.UseTranslationsApi)
```
